### PR TITLE
Use our own libarchive in lanraragi

### PR DESCRIPTION
**Pull request by fxcoudert**
See https://github.com/Homebrew/homebrew-core/pull/183626 and https://github.com/Homebrew/homebrew-core/pull/181569, there are good reasons to use our own libarchive everywhere rather than depend on Apple's:

- consistency
- Apple's version lacks headers, which makes formulas more complex
- Apple's version is old, provided only for backward compatibility, and lacks some functionality

### CODE REVIEW

**Review by carlocab — APPROVED** [review]

File: audit_exceptions/provided_by_macos_depends_on_allowlist.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [
   "apr",
   "apr-util",
+  "libarchive",
```
Comment (bugdea1er) [inline audit_exceptions/provided_by_macos_depends_on_allowlist.json:4]: I don't think that's enough, Homebrew should explicitly disallow `uses_from_macos libarchive`?

Comment (fxcoudert) [reply audit_exceptions/provided_by_macos_depends_on_allowlist.json:4]: At this stage there is no reason, and it would be a brew change.

**Review by MikeMcQuaid — APPROVED** [review]
Makes sense!

File: audit_exceptions/provided_by_macos_depends_on_allowlist.json (hunk 1 above)
Comment (MikeMcQuaid) [reply audit_exceptions/provided_by_macos_depends_on_allowlist.json:4]: @fxcoudert Should this `brew` change not happen right now or ever? Can you elaborate a bit more on reasoning here? Can be post-merge.

Comment (carlocab) [reply audit_exceptions/provided_by_macos_depends_on_allowlist.json:4]: There's an open issue for this at https://github.com/Homebrew/brew/issues/18264, let's continue discussion there.

Marking this as resolved to allow this PR to merge.

**Pull request merged**